core/src/main/java/hudson/Util.java

```diff
@@ -1180,14 +1177,16 @@ public static void touch(@Nonnull File file) throws IOException {
 
     /**
      * Copies a single file by using Ant.
+     *
+     * @deprecated use {@link Files#copy(Path, Path, CopyOption...)}
```

Comment:
My only hesitation for deprecating the method is that handling the InvalidPathException from the File to Path conversion can be annoying for external callers who don't have access to `Util#fileToPath`. I don't feel too strongly either way though.